packages/common-utils/src/queryParser.ts

```diff
@@ -666,50 +726,88 @@ async function nodeTerm(
       rangedTerm.term_min,
       rangedTerm.term_max,
       isNegatedField,
+      context,
     );
   }
 
   throw new Error(`Unexpected Node type. ${node}`);
 }
 
+function createSerializerContext(
+  currentContext: SerializerContext,
+  ast: lucene.BinaryAST | lucene.LeftOnlyAST,
+) {
+  // For syntax like `foo:(bar baz)` or `foo:("bar baz")`, the implicit field for the inner expression must be `foo`
+  if (ast.field && ast.parenthesized && ast.field !== IMPLICIT_FIELD) {
+    const fieldWithoutNegation = ast.field?.startsWith('-')
+      ? ast.field.slice(1)
+      : ast.field;
+
+    return {
+      ...currentContext,
+      implicitColumnExpression: fieldWithoutNegation,
+    };
+  } else {
+    return currentContext;
+  }
+}
+
 async function serialize(
   ast: lucene.AST | lucene.Node,
   serializer: Serializer,
+  context: SerializerContext,
 ): Promise<string> {
   // Node Scenarios:
   // 1. NodeTerm: Single term ex. "foo:bar"
   // 2. NodeRangedTerm: Two terms ex. "foo:[bar TO qux]"
   if ((ast as lucene.NodeTerm).term != null) {
-    return await nodeTerm(ast as lucene.NodeTerm, serializer);
+    return await nodeTerm(ast as lucene.NodeTerm, serializer, context);
   }
   if ((ast as lucene.NodeRangedTerm).inclusive != null) {
-    return await nodeTerm(ast as lucene.NodeTerm, serializer);
+    return await nodeTerm(ast as lucene.NodeTerm, serializer, context);
   }
 
   // AST Scenarios:
   // 1. BinaryAST: Two terms ex. "foo:bar AND baz:qux"
   // 2. LeftOnlyAST: Single term ex. "foo:bar"
   if ((ast as lucene.BinaryAST).right != null) {
     const binaryAST = ast as lucene.BinaryAST;
-    const operator = serializer.operator(binaryAST.operator);
+    const operator = serializer.operator(binaryAST.operator, context);
     const parenthesized = binaryAST.parenthesized;
-    return `${parenthesized ? '(' : ''}${await serialize(
+
+    // Handle cases like "-foo:(bar baz)"
+    const isNegatedAndParenthesizedField =
+      parenthesized && binaryAST.field?.startsWith('-');
```

Comment:
style: This code snippet appears in several places. We can probably add a AST helper to do so